crates/coordinate_systems/src/framed_nalgebra.rs

```diff
@@ -0,0 +1,303 @@
+
+impl<Frame, const DIMENSION: usize> Point<Frame, DIMENSION> {
+    pub fn origin() -> Self {
+        Self::wrap(nalgebra::Point::origin())
+    }
+
+    pub fn coords(&self) -> Framed<Frame, SVector<f32, DIMENSION>> {
+        Framed::wrap(self.inner.coords)
+    }
+}
+
+impl<Frame, From, const DIMENSION: usize> Point<Frame, DIMENSION, From>
+where
+    From: Scalar,
+{
+    pub fn map<F, To>(&self, f: F) -> Point<Frame, DIMENSION, To>
+    where
+        To: Scalar,
+        F: FnMut(From) -> To,
+    {
+        Framed::wrap(self.inner.map(f))
+    }
+}
+
+impl<Frame, Scalar> Point2<Frame, Scalar>
+where
+    Scalar: nalgebra::Scalar + Zero + One + Copy,
+{
+    pub fn x(&self) -> Scalar {
+        self.inner.x
+    }
+
+    pub fn y(&self) -> Scalar {
+        self.inner.y
+    }
+}
+
+impl<Frame, Scalar> Point3<Frame, Scalar>
+where
+    Scalar: nalgebra::Scalar + Zero + One + Copy,
+{
+    pub fn x(&self) -> Scalar {
+        self.inner.x
+    }
+
+    pub fn y(&self) -> Scalar {
+        self.inner.y
+    }
+
+    pub fn z(&self) -> Scalar {
+        self.inner.z
+    }
+}
+
+impl<Frame, const DIMENSION: usize, Scalar> From<nalgebra::SVector<Scalar, DIMENSION>>
+    for Point<Frame, DIMENSION, Scalar>
+where
+    Scalar: nalgebra::Scalar,
+{
+    fn from(value: nalgebra::SVector<Scalar, DIMENSION>) -> Self {
+        Self::wrap(value.into())
+    }
+}
+
+// Pose
+
+impl<Frame> Pose<Frame> {
+    pub fn new(translation: Vector2<Frame, f32>, angle: f32) -> Self {
+        Pose::wrap(nalgebra::Isometry2::new(translation.inner, angle))
+    }
+
+    pub fn from_parts(translation: Vector2<Frame, f32>, orientation: Orientation<Frame>) -> Self {
+        Pose::wrap(nalgebra::Isometry2::from_parts(
+            translation.inner.into(),
+            orientation.inner,
+        ))
+    }
+
+    pub fn from_position(position: Point2<Frame>) -> Self {
+        Pose::wrap(nalgebra::Isometry::from(position.inner))
+    }
+
+    pub fn as_transform<From>(&self) -> Isometry2<From, Frame> {
+        Isometry2::wrap(self.inner)
+    }
+
+    pub fn position(&self) -> Point2<Frame> {
+        Point2::wrap(self.inner.translation.vector.into())
+    }
+
+    pub fn orientation(&self) -> Orientation<Frame> {
+        Orientation::wrap(self.inner.rotation)
+    }
+
+    pub fn angle(&self) -> f32 {
+        self.inner.rotation.angle()
+    }
+}
+
+// Orientation
+
+impl<Frame> Orientation<Frame> {
+    pub fn new(angle: f32) -> Self {
+        Self::wrap(nalgebra::UnitComplex::new(angle))
+    }
+
+    pub fn identity() -> Self {
+        Self::wrap(nalgebra::UnitComplex::identity())
+    }
+
+    pub fn from_cos_sin_unchecked(cos: f32, sin: f32) -> Self {
+        Self::wrap(nalgebra::UnitComplex::from_cos_sin_unchecked(cos, sin))
+    }
+
+    pub fn from_vector(direction: Vector2<Frame>) -> Self {
+        Orientation::wrap(nalgebra::UnitComplex::rotation_between(
+            &nalgebra::Vector2::x_axis(),
+            &direction.inner,
+        ))
+    }
```

Comment:
This implication is what defines this whole type, its not just a property of this constructor.
This is how all our poses have worked since forever: If the rotational part is an identity mapping, the forward vector aligns with the X axis.